cmd/tools/grafana/dashboard_test.go

```diff
@@ -12,8 +13,80 @@ import (
 
 var dashboards = []string{"../../../grafana/dashboards/cmode", "../../../grafana/dashboards/storagegrid"}
 
+func TestThreshold(t *testing.T) {
+	visitDashboards(dashboards, func(path string, data []byte) {
+		checkThreshold(t, path, data)
+	})
+}
+
+func checkThreshold(t *testing.T, path string, data []byte) {
+	path = shortPath(path)
+	var thresholdMap = map[string][]string{
+		"_latency": {
+			"[\"green\",\"orange\",\"red\"]",
+			"[null,20,30]",
+		},
+		"_busy": {
+			"[\"green\",\"orange\",\"red\"]",
+			"[null,60,80]",
+		},
+	}
+	// visit all panel for datasource test
+	visitAllPanels(data, func(p string, key, value gjson.Result) {
+		panelTitle := value.Get("title").String()
+		kind := value.Get("type").String()
+		if kind == "table" {
+			targetsSlice := value.Get("targets").Array()
+			for _, targetN := range targetsSlice {
+				expr := targetN.Get("expr").String()
+				if strings.Contains(expr, "_latency") || strings.Contains(expr, "_busy") {
+					var th []string
+					if strings.Contains(expr, "_latency") {
+						th = thresholdMap["_latency"]
+					} else if strings.Contains(expr, "_busy") {
+						th = thresholdMap["_busy"]
+					}
+					isThresholdSet := false
+					isColorBackgroundSet := false
+					expectedColorBackground := []string{"color-background", "lcd-gauge"}
+					// check if any override has threshold set
+					overridesSlice := value.Get("fieldConfig.overrides").Array()
+					for _, overrideN := range overridesSlice {
+						propertiesSlice := overrideN.Get("properties").Array()
+						for _, propertiesN := range propertiesSlice {
+							id := propertiesN.Get("id").String()
+							if id == "thresholds" {
+								color := propertiesN.Get("value.steps.#.color")
+								v := propertiesN.Get("value.steps.#.value")
+								if color.String() == th[0] {
+									if v.String() == th[1] {
+										isThresholdSet = true
```

Comment:
minor: could be simplified to `isThresholdSet = color.String() == th[0] && v.String() == th[1]`